src/commands/apply-chain-updates.ts

```diff
@@ -0,0 +1,372 @@
+ * in ASCII text strings of 32-44 characters. This function only validates the character
+ * pattern and length.
+ *
+ * @warning Since this function depends on user's input only, it:
+ * - DOES NOT validate if the address is a valid ed25519 public key
+ * - DOES NOT validate checksum (typos cannot be detected)
+ * - Single character typos, reversed characters, and case errors may still pass
+ *
+ * @param address - A string representing a Solana address in Base58 format.
+ * @returns {boolean} - Returns true if the address is valid, false otherwise.
+ */
+function isSolanaAddress(address: string): boolean {
+  const SOLANA_ADDRESS_REGEX = /^[1-9A-HJ-NP-Za-km-z]{32,44}$/
+
+  return SOLANA_ADDRESS_REGEX.test(address)
+}
+
+/**
+ * @description Converts a Solana address from Base58 format to a 32-byte hex string which CCIP expects.
+ *
+ * If the address is not valid or cannot be decoded to 32 bytes, it returns the original address.
+ *
+ * @param address - A string representing a Solana address in Base58 format.
+ * @returns {string} - Returns the address as a 32-byte hex string prefixed with '0x'.
+ */
+function convertSolanaAddressToBytes32Hex(address: string): string {
+  try {
+    const decoded = bs58.decode(address)
+    if (decoded.length !== 32) {
+      console.warn(
+        `⚠️  Invalid Solana address: "${address}" decoded to ${decoded.length} bytes, expected 32 bytes.`,
+      )
+      return address // Return original address if length is not 32 bytes
+    } else {
+      const hexAddress = '0x' + Buffer.from(decoded).toString('hex')
+      console.warn('Converted Solana Base58 address to 32 bytes hex:', hexAddress)
+      return hexAddress
+    }
+  } catch (error) {
+    console.warn(`⚠️  Error converting Solana address "${address}":`, error)
+    return address // Return original address in case of error
+  }
+}
+
+/**
+ * @description Generates calldata for the `applyChainUpdates` function of the TokenPool.sol smart contract.
+ *
+ * This function is useful if you work with multisig wallets.
+ *
+ * @param argv.json_args - Path to a JSON file containing chain updates arguments.
+ * @param argv.format - The format in which to display the output (log, pretty, or json).
+ * @returns {calldata: string, remoteChainSelectorsToRemove: bigint[], chainsToAdd: ChainToAdd[]}
+ */
+export async function generateApplyChainUpdatesCalldata(argv: {
+  json_args: string // Path to a JSON file containing chain updates arguments
+  format: Format
+}): Promise<{
+  calldata: string
+  remoteChainSelectorsToRemove: bigint[]
+  chainsToAdd: ChainToAdd[]
+}> {
+  // ================================================================
+  // │              Extract Arguments from JSON File                │
+  // ================================================================
+  const argsContent = await readFile(argv.json_args, 'utf-8')
+  const args = JSON.parse(argsContent)
+
+  const remoteChainSelectorsToRemove = Array.isArray(args)
```

Comment:
At this stage `const remoteChainSelectorsToRemove = Array.isArray(args)` -- wont args always be an object?  the example json file shows it as always an object?

Reply:
right but isnt args in line 97 an object? its reading the whole file (which is the example file you added in this PR   https://github.com/smartcontractkit/ccip-tools-ts/pull/40/files#diff-7d9542aa81f373a120e8c1fea4b347c033a8ca62ff64b319c3f1ebb0db46aaac)

which means `remoteChainSelectorsToRemove` is a property on that object.   Currently the Array.isArray() will always be false when its `args` which is an object?  

I'll leave it to you to and @andrevmatos  -- but my suggestion is to require that the user only provides an object in the file (i.e. the file must never just be a JSON array) so we simple read `remoteChainSelectorsToRemove` and `chainsToAdd` from that object in which case the ternary check on  line 99 wont be needed etc.